components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -120,3 +148,59 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 		return r.updateDeploymentStatus(ctx, log, instance, deployments.Items, corev1.ConditionTrue)
 	}
 }
+
+func isOnSourceChange(instance *serverlessv1alpha1.Function, commit string) bool {
+	return instance.Status.Commit == "" ||
+		commit != instance.Status.Commit ||
+		instance.Spec.Branch != instance.Status.Branch ||
+		instance.Spec.Runtime != instance.Status.Runtime ||
+		instance.Spec.BaseDir != instance.Status.BaseDir
+}
+
+func (r *FunctionReconciler) onSourceChange(ctx context.Context, instance *serverlessv1alpha1.Function, repository *serverlessv1alpha1.Repository) (ctrl.Result, error) {
+	return r.updateStatus2(ctx, ctrl.Result{}, instance, serverlessv1alpha1.Condition{
+		Type:               serverlessv1alpha1.ConditionConfigurationReady,
+		Status:             corev1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+		Reason:             serverlessv1alpha1.ConditionReasonSourceUpdated,
+		Message:            fmt.Sprintf("Sources %s updated", instance.Name),
+	}, repository)
+}
+
+var gitOperator = git.New()
```

Comment:
gitOperator is meant to be a global? or do you want to refactor it with the rest, in next PR?

Reply:
It will be fixed in upcoming PR, thx.